edn/tests/tests.rs

```diff
@@ -17,9 +17,22 @@ use std::iter::FromIterator;
 use num::bigint::ToBigInt;
 use num::traits::{Zero, One};
 use ordered_float::OrderedFloat;
+use edn::keyword;
+use edn::types::Value;
 use edn::types::Value::*;
 use edn::parse::*;
 
+// Helper for making wrapped keywords with a namespace.
+fn k_ns(ns: &str, name: &str) -> Value {
```

Comment:
We can't use `From`: those conversions cannot fail, and not all strings can be turned into a keyword.

Indeed, even `TryFrom` returns a `Result`, and thus complicates matters.

This is the root of parsing, and why I picked this division: the struct and its constructors aren't failable (they include panicking validation, but that's another matter), whereas the parser itself will refuse to parse a malformed keyword.